File: src/main/java/se/augustoccesar/querybuilder/query/trackers/OrderTracker.java
package se.augustoccesar.querybuilder.query.trackers;

import se.augustoccesar.querybuilder.builders.Buildable;
import se.augustoccesar.querybuilder.constants.CommonStrings;
import se.augustoccesar.querybuilder.query.Order;

import java.util.ArrayList;
import java.util.Arrays;

/**
 * Author: augustoccesar
 * Date: 10/05/17
 */
public class OrderTracker implements Buildable {
    private ArrayList<Order> orders = new ArrayList<>();

    public OrderTracker addOrder(Order order) {
        this.orders.add(order);
        return this;
    }

    public OrderTracker addOrders(Order... orders) {
        this.orders.addAll(Arrays.asList(orders));
        return this;
    }

    @Override
    public String build() {
        StringBuilder stringBuilder = new StringBuilder();

        for (int i = 0; i < this.orders.size(); i++) {
            stringBuilder.append(this.orders.get(i).build());
            if (i < this.orders.size() - 1) {
                stringBuilder.append(CommonStrings.COMMA);
            }
        }

        return stringBuilder.toString().replaceAll("\\s+", " ");
    }

    @Override
    public boolean shouldBuild() {
        return orders.size() > 0;
    }
}

File: src/main/java/br/com/augustoccesar/querybuilder/query/trackers/OrderTracker.kt
package br.com.augustoccesar.querybuilder.query.trackers

import br.com.augustoccesar.querybuilder.builders.Buildable
import br.com.augustoccesar.querybuilder.constants.CommonStrings
import br.com.augustoccesar.querybuilder.query.Order
import java.util.*

/**
 * Author: augustoccesar
 * Date: 10/05/17
 */
class OrderTracker : Buildable {
    private val orders = ArrayList<Order>()

    fun addOrder(order: Order): OrderTracker {
        this.orders.add(order)
        return this
    }

    fun addOrders(vararg orders: Order): OrderTracker {
        this.orders.addAll(Arrays.asList(*orders))
        return this
    }

    override fun build(): String {
        val stringBuilder = StringBuilder()

        for (i in this.orders.indices) {
            stringBuilder.append(this.orders[i].build())
            if (i < this.orders.size - 1) {
                stringBuilder.append(CommonStrings.COMMA)
            }
        }

        return stringBuilder.toString().replace("\\s+".toRegex(), " ")
    }

    override fun shouldBuild(): Boolean {
        return orders.size > 0
    }
}
